lms/static/js/edxnotes/views/tabs/tags.js

```diff
@@ -0,0 +1,129 @@
+                    // We can assume it would be the last element of the list because we iterate through
+                    // all tags on a given note before moving on to the text note.
+                    if (noteList.length === 0 || noteList[noteList.length - 1] !== note) {
+                        noteList.push(note);
+                    }
+                };
+
+                this.collection.each(function(note){
+                    tags = note.get('tags');
+                    if (tags.length === 0) {
+                        addNoteForTag(note, noTags);
+                    }
+                    else {
+                        for (i = 0; i < tags.length; i++) {
+                            addNoteForTag(note, tags[i]);
+                        }
+                    }
+                });
+
+                sortedTagNames = Object.keys(notesByTag).sort(function (a, b) {
+                    // "no tags" should always appear last
+                    if (a === noTags) {
+                        return 1;
+                    }
+                    else if (b === noTags) {
+                        return -1;
+                    }
+                    else if (notesByTag[a].length > notesByTag[b].length) {
+                        return -1;
+                    }
+                    else if (notesByTag[a].length < notesByTag[b].length) {
+                        return 1;
+                    }
+                    else {
+                        return a.toLowerCase() <= b.toLowerCase() ? -1 : 1;
+                    }
+                });
+
+               container = document.createDocumentFragment();
+
+                // Store map of titles for scrollToTag functionality.
+                this.titleMap = {};
+                titleMap = this.titleMap;
+
+                _.each(sortedTagNames, function (tagName) {
+                    noteGroup = notesByTag[tagName];
+                    var tagTitle = interpolate_text(
+                        gettext("{tagName} ({numberOfNotesWithTag})"),
+                        {tagName: tagName, numberOfNotesWithTag: noteGroup.length}
+                    );
+                    group = this.getGroup(tagTitle);
+                    titleMap[tagName] = tagTitle;
+
+                    group.addChild(this.getNotes(noteGroup));
+                    container.appendChild(group.render().el);
+                }, this);
+
+                this.$el.append(container);
+                return this;
+            },
+
+            getGroup: function (tagName) {
+                var group = new NoteGroupView.GroupView({
+                    displayName: tagName,
+                    template: '<h3 class="tags-title"><%- displayName %></h3>',
+                    className: "note-group"
+                });
+                this.children.push(group);
+                return group;
+            }
+        }),
+
+        tabInfo: {
+            name: gettext('Tags'),
```

Comment:
Frankly I'd provide context here. a "tag" could mean a whole lot of different things (price tag, website label tags, graffiti or even "tag" as a verb - see how many definitions are here! http://dictionary.reference.com/browse/tag)